### PR TITLE
feat(FT-4738): Added suppliers pull capabilities to Xero

### PR DESCRIPTION
**Pull request by SvetlinStaevPayhawk**

### CODE REVIEW

File: src/managers/integration/Manager.ts [hunk 1]
```diff
@@ -138,6 +162,69 @@ export class Manager implements IManager {
...
+        const xeroVendors = xeroContacts.map(vendor => ({
+            id: vendor.contactId,
+            name: vendor.name,
+            vat: vendor.taxNumber,
+            customFields: [],
+        }));
+
+        const vendors = xeroVendors.filter(vendor => !!vendor.name?.trim()).sortBy(v => v.id);
+        const uniqueVendorsByName = filterUniqBy<IVendor>(vendors, ['name']);
+        const uniqueVendorsByNameAndVat = filterUniqBy<IVendor>(uniqueVendorsByName, ['vat']);
```
Comment (vassildinev) [inline src/managers/integration/Manager.ts:194]: can this be extracted in utils package?

File: src/services/payhawk/Client.ts [hunk 2]
```diff
@@ -216,4 +219,31 @@ export class Client implements IClient {
 
         return result;
     }
+
+    async replaceSupplierCustomFields(payload: V3.INewSupplierCustomFields) {
```
Comment (vassildinev) [inline src/services/payhawk/Client.ts:223]: let's add a comment on these why they are the only ones on v3

File: src/managers/xero-entities/Manager.ts [hunk 3]
```diff
@@ -46,6 +46,10 @@ export class Manager implements IManager {
         return this.xeroClient.accounting.getTrackingCategories();
     }
 
+    async getContacts(): Promise<IContact[]> {
+        return await this.xeroClient.accounting.getContacts();
+    }
```
Comment (vassildinev) [inline src/managers/xero-entities/Manager.ts:51]: is this needed? we are using the AccountingClient, right?

File: src/shared/client-contracts/Contact.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+export interface IContact {
```
Comment (vassildinev) [inline src/shared/client-contracts/Contact.ts:1]: can we reuse the V3 contracts, I 'm not sure where these are used

File: src/managers/integration/Manager.ts (hunk 1 above)
Comment (SvetlinStaevPayhawk) [reply src/managers/integration/Manager.ts:194]: Yeah, both this operation and possibly the supplierPayload can be extracted in a builder in the utils.

File: src/managers/xero-entities/Manager.ts (hunk 3 above)
Comment (SvetlinStaevPayhawk) [reply src/managers/xero-entities/Manager.ts:51]: Yes, this is executed through the Accounting Client, I added it to keep up with the convention used in the `getTaxRates` and `getTrackingCategories` methods in the same class.

File: src/shared/client-contracts/Contact.ts (hunk 4 above)
Comment (SvetlinStaevPayhawk) [reply src/shared/client-contracts/Contact.ts:1]: For the `IVendor` and the related interfaces we can do that but the `IContact` is the contact with the Xero API, is it not the right place to carry it at?

Comment (vassildinev) [reply src/shared/client-contracts/Contact.ts:1]: yes, this one is OK here, I meant the others

**Review by vassildinev — APPROVED** [review]

**Pull request merged**